lib/appium.js

```diff
@@ -165,37 +175,47 @@ class AppiumDriver extends BaseDriver {
 
     // sessionOverride server flag check
     // this will need to be re-thought when we go to multiple session support
-    if (this.args.sessionOverride && !!this.sessions && _.keys(this.sessions).length > 0) {
-      log.info('Session override is on. Deleting other sessions.');
-      for (let id of _.keys(this.sessions)) {
-        log.info(`    Deleting session '${id}'`);
-        try {
-          await this.deleteSession(id);
-        } catch (ign) {
-          // the error has already been logged in AppiumDriver.deleteSession
-          // continue
+    if (this.args.sessionOverride) {
+      const sessionIdsToDelete = await sessionsListGuard.acquire(AppiumDriver.name, () => _.keys(this.sessions));
```

Comment:
just because we pass `this.sessions` to a method doesn't mean it has to be used. the other instances would be simplified because they wouldn't require remembering the lock name:

```js
await this.guardedSessions(() => {
  // mutate this.sessions in here or do whatever
});
```

is still a bit nicer looking IMO.

this isn't a big deal, i just habitually look for ways to remove duplication :-)